src/PrestaShopBundle/Twig/Component/QuickAccess.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Twig\Component;
+
+use Link;
+use Symfony\UX\TwigComponent\Attribute\AsTwigComponent;
+
+#[AsTwigComponent(template: '@PrestaShop/Admin/Component/Layout/quick_access.html.twig')]
+class QuickAccess
+{
+    public array $quickAccess;
+    public Link $link;
```

Comment:
Is it possible to use something else than the legacy class ? 
I'd try to avoid legacy in the new twig templates

Reply:
That's the purpose, the component creations is divided into two steps:
- first create the component with dependencies injected via some variables injected by smarty
- then make the component independent thanks to DI and remove all legacy dependencies (create them if necessary)

That's why Salim created a dedicated PHPStan exceptions file that is temporary and will be clean piece by piece during phase 2.

You can find more details in this Epic https://github.com/PrestaShop/PrestaShop/issues/33101 (phase 2 issues still need to be created but it's mainly for project management in the Kanban)